persistence/trees/trees.go

```diff
@@ -71,26 +71,29 @@ type stateTree struct {
 	nodeStore kvstore.KVStore
 }
 
-var _ modules.TreeStoreModule = &treeStore{}
+var _ modules.TreeStoreModule = &TreeStore{}
 
-// treeStore stores a set of merkle trees that
+// TreeStore stores a set of merkle trees that
 // it manages. It fulfills the modules.TreeStore interface.
 // * It is responsible for atomic commit or rollback behavior
 // of the underlying trees by utilizing the lazy loading
 // functionality provided by the underlying smt library.
-type treeStore struct {
+type TreeStore struct {
```

Comment:
Given that we have an interface `TreeStoreModule`, we shouldn't need to publically expose the implementation as well. Otherwise, feels like we're not encapsulating/defining logic correctly.

1. Why not update TreeStoreModule?
2. What needs to happen to not expose this?

Reply:
This is necessary for testing. If we don't do this, then we can't access the right methods, and if we keep the tests in the same package as trees, then we get import errors because we haven't ironed out some of the dependencies.

Reply:
In the future, I think that if we're making a design decision solely for testing, we should add a `TECHDEBT` comment that says something along the lines of "exposed for testing purposes.

---

> This is necessary for testing. If we don't do this, then we can't access the right methods

Can you look at `ConsensusDebugModule` and expose the functions we need in a testing interface with a `TECHDEBT(#880)` comment above it?

> then we get import errors because we haven't ironed out some of the dependencies.

Can you add details on the import errors in #880? I'm guessing they're related. cc @bryanchriswhite

Reply:
If testing is the only place where this would be imported you can add a file to this package which includes the `//go:build test` build tag and exports it for usage in tests:

```
type TreeStore = treeStore

func (ts *treeStore) SomeTestHelperMethod(...) { ... }
```

See [p2p/raintree/testutil.go](https://github.com/pokt-network/pocket/blob/main/p2p/raintree/testutil.go) or [p2p/testutil.go](https://github.com/pokt-network/pocket/blob/main/p2p/testutil.go) for examples.

Reply:
> Can you add details on the import errors in #880? I'm guessing they're related. cc @bryanchriswhite

:100: :pray:

Reply:
@Olshansk I added a techdebt comment linking to #880 and will include the test tag changes in a future PR that incorporates the tests. The capitalization is in this PR because I split the branch into multiple PRs after I had written the tests.

@bryanchriswhite I'll include the tag and custom struct technique in the [TestUtils doc](https://www.notion.so/pocketnetwork/Testutils-9cba9010e18447248e9daa8a3b87e3f2) as well, because it's mentioned but not detailed and it's a smart trick to get around import errors for now until we fix the deeper issue, and might still be useful later on anyway.

Reply:
@dylanlott I created a branch #920 and ran the tests and everything worked out fine.

Where are the tests that are blocking this?

Reply:
They're in a WIP PR that was based off of the branch that I split up to make these first three commits, so this will work right now, and I'll use the approach @bryanchriswhite suggested to adjust the package scope for the names later, that works fine by me.